### ISSUE TITLE
http lambda native doesn't work as expected

### ISSUE THREAD

**Issue by VSVDEv** [quarkusio/quarkus#44290]
### Describe the bug

I create sample project using https://quarkus.io/guides/aws-lambda-http
When I build and deployed native i got init time and I see that is not native

### Expected behavior

work without init phase with execution 15 ms

### Actual behavior

log

```

  | 2024-11-04T13:39:26.554Z | INIT_START Runtime Version: provided:al2023.v40 Runtime Version ARN: arn:aws:lambda:us-east-1:3
-- | -- | --
  | 2024-11-04T13:39:26.944Z | __ ____ __ _____ ___ __ ____ ______
  | 2024-11-04T13:39:26.944Z | --/ __ \/ / / / _ \| / _ \/ //_/ / / / __/
  | 2024-11-04T13:39:26.944Z | -/ /_/ / /_/ / __ \|/ , _/ ,< / /_/ /\ \
  | 2024-11-04T13:39:26.944Z | --\___\_\____/_/ \|_/_/\|_/_/\|_\|\____/___/
  | 2024-11-04T13:39:26.944Z | 2024-11-04 13:39:26,944 INFO [io.quarkus] (main) ntest 0.0.1 native (powered by Quarkus 3.15.1) started in 0.190s.
  | 2024-11-04T13:39:26.944Z | 2024-11-04 13:39:26,944 INFO [io.quarkus] (main) Profile prod activated.
  | 2024-11-04T13:39:26.944Z | 2024-11-04 13:39:26,944 INFO [io.quarkus] (main) Installed features: [amazon-lambda, cdi, resteasy, security, smallrye-context-propagation, vertx]
  | 2024-11-04T13:39:26.946Z | START RequestId: da011a65-4cf9-4129-b879-9e098136355e Version: $LATEST
  | 2024-11-04T13:39:27.385Z | END RequestId: da011a65-4cf9-4129-b879-9e098136355e
  | 2024-11-04T13:39:27.385ZREPORT RequestId: da011a65-4cf9-4129-b879-9e098136355e	Duration: 439.06 ms	Billed Duration: 831 ms	Memory Size: 128 MB	Max Memory Used: 67 MB	Init Duration: 391.13 ms

```

### How to Reproduce?

here is app https://github.com/VSVDEv/nativetest
use steps to build and deploy https://github.com/VSVDEv/nativetest/blob/master/HELP.md 

### Output of `java -version`

17.0.2

### Quarkus version or git rev

3.15.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

3.9.2

**Comment by Karm** [comment]
Hello @VSVDEv, could you rephrase what you perceive as wrong here? I can see in the log you posted that:

* 13:39:26.554 your AWS environment woke up
* 13:39:26.944 Quarkus native started to print its banner.  Was there any log before that? What was happening for ~400ms before that?
*  Quarkus started in ~190ms as it reports
* 13:39:26.946 Processing of a request started
* 13:39:27.385 processing of a request ended

What are you worried about? The only thing that seems fishy to me is that there is a ~400ms gap between the env logs  INIT_START and when Quarkus reports it has started in ~190ms , so something happened before that that took ~200ms which is a long time. I don't know why it would be a Quarkus issue though. Could it be the AWS env loading something?

**Comment by VSVDEv** [comment]
Hi here is a spring issue https://github.com/aws/serverless-java-container/issues/1087 with native image. It doesn't have init duration(you can find the last logs "Report") 
REPORT RequestId: b8b15b84-8696-497c-bc83-751c82bf6e4b Duration: 10.70 ms Billed Duration: 11 ms Memory Size: 512 MB Max Memory Used: 138 MB
The Duration: 10.70 ms when I use quarkus app this example  I got over 100 ms without additional dependencies. And if you take pause 5-7 min spring has The Duration: 10-20 ms, quarkus has again init duration. So i believe it create fallback image. When I use option to build in container how can I prevent fallback image?

**Comment by zakkak** [comment]
FWIW Quarkus [explicitly disables fallback images by default](https://quarkus.io/guides/all-config#quarkus-core_quarkus-native-enable-fallback-images). So if you are not setting `quarkus.native.enable-fallback-images` or `QUARKUS_NATIVE_ENABLE_FALLBACK_IMAGES` in your builds, creating a fallback image shouldn't be possible.

**Comment by zakkak** [comment]
> It doesn't have init duration(you can find the last logs "Report")

I guess you are referring to the logs in the [issue description](https://github.com/aws/serverless-java-container/issues/1087#issue-2602004205). AFAIU this is because the app fails during initialization and the init duration is reported in the `INIT_REPORT` in that case, see https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html#runtimes-lifecycle-init-errors.

If you inspect the logs in https://github.com/aws/serverless-java-container/issues/1087#issuecomment-2451657432 you see that init duration is included in the `REPORT` there as well:

```
REPORT RequestId: 6913046c-15b2-4956-8409-22788a7e5704 Duration: 205.13 ms Billed Duration: 957 ms Memory Size: 512 MB Max Memory Used: 134 MB Init Duration: 751.82 ms
```

**Comment by andrezimmermann** [comment]
I'm not a Quarkus dev, but I think is okay for me to chime in.

> It doesn't have init duration(you can find the last logs "Report")

If you look at the Spring Boot logs, they do include INIT phase. It is impossible not have one since is required as part of the AWS Lambda lifecycle

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html

I will copy and paste from the attached logs:

```
REPORT RequestId: 6913046c-15b2-4956-8409-22788a7e5704 Duration: 205.13 ms Billed Duration: 957 ms Memory Size: 512 MB Max Memory Used: 134 MB Init Duration: 751.82 ms
```

Also, while is not obvious memory size of the lambda also dictates the CPU speed. A fair comparison would be having the Quarkus lambda and Spring Boot lambda with the same memory configuration.

https://docs.aws.amazon.com/lambda/latest/operatorguide/computing-power.html

**Comment by VSVDEv** [comment]
@zakkak, @andrezimmermann  yes but I would like to mention that "Init Duration" included only once I tested both app in 1 minute, in 3 minute, in 5 minute, in 10 minute. So in 1 and 3 minute lambda doesn't have "Init duration" even not native app. In 5 minutes jvm app(quarkus, spring) has "Init duration". In 5minutes jvm app(quarkus, spring) with snap start has "restore duration".  In 5 minutes native app spring doesn't have "Init duration".  In 5 minutes native app quarkus has  "Init duration". That is the problem what I 'm talking about. And also you can see that basic app quarkus with only one hello resource has Duration 10 times bigger than spring app which call code to retrieve from DynamoDb table.

**Comment by andrezimmermann** [comment]
If I understood correctly, you sent a HTTP request once, then after 1 minute, then again in 3 minutes, then again in 5 minutes, lastly after 10 minutes?

>  In 5 minutes native app quarkus has "Init duration". That is the problem what I 'm talking about.

Neither Quarkus or Spring have any configuration, setting or say in how AWS controls the lifecycle of a AWS Lambda.
If this is root of the issue, this is how AWS Lambda works.

> And also you can see that basic app quarkus with only one hello resource has Duration 10 times bigger than spring app which call code to retrieve from DynamoDb table.

From the information that was shared, the quarkus function with 128 MB was billed 831 ms and the Spring function with 512 MB that was billed 957 ms on the initial request.

There is no logs for the warm invocations so I can't compare them.

> Quarkus
> REPORT RequestId: da011a65-4cf9-4129-b879-9e098136355e	Duration: 439.06 ms	Billed Duration: 831 ms	Memory Size: 128 MB	Max Memory Used: 67 MB	Init Duration: 391.13 ms

> Spring
> REPORT RequestId: 6913046c-15b2-4956-8409-22788a7e5704 Duration: 205.13 ms Billed Duration: 957 ms Memory Size: 512 MB Max Memory Used: 134 MB Init Duration: 751.82 ms

In the end this is comparing apples and oranges since the memory size is different, hence different processing power. 

I'm sorry if I sound combative, that is not my intent, I extensively use Quarkus with native binaries with AWS Lambda and the mentioned behavior seems expected. I considered using Spring Boot too but my findings were that it was slower than Quarkus overall. The only benefit was the skill transfer of the Spring Boot expertise into micro service development.
Maybe I'm missing out on something? Perhaps someone else will have a different point of view.

**Comment by VSVDEv** [comment]
> ntest-0.0.1-runner
how can I see if it's a fallback or not? As result I have: ntest-0.0.1-runner, ntest-0.0.1.jar, function.zip
function.zip contains bootstrap file

**Comment by zakkak** [comment]
If `ntest-0.0.1-runner` is a binary file (can be seen with `file ntest-0.0.1-runner`) and contains symbol names like `__svm_vm_java_version` then it's not a fallback. I repeat that Quarkus doesn't generate fallbacks in any case.

**Comment by Karm** [comment]
@VSVDEv  Hello, I went over this again and I'm closing it as Quarkus unrelated. 
All lambdas have init duration as debated above. 

If you can pinpoint a specific issue, e.g. Native image starting suspiciously long or consuming suspicious amount of resources, please, reopen.

**Comment by VSVDEv** [comment]
@Karm  I believe I found that problem in memory size when I increased to 512 MB It has similar parameters as spring app(lost init duration after 5 minutes). I think you can add to the documentation that generated sam.native.yaml by default has only 128 MB, sam.jvm.yaml has by default 512 MB
